datafusion/functions-array/src/planner.rs

```diff
@@ -97,6 +98,27 @@ impl ExprPlanner for ArrayFunctionPlanner {
     ) -> Result<PlannerResult<Vec<Expr>>> {
         Ok(PlannerResult::Planned(make_array(exprs)))
     }
+
+    fn plan_make_map(&self, args: Vec<Expr>) -> Result<PlannerResult<Vec<Expr>>> {
+        if args.len() % 2 != 0 {
+            return exec_err!("make_map requires an even number of arguments");
+        }
+
+        let (keys, values): (Vec<_>, Vec<_>) = args
+            .chunks_exact(2)
+            .map(|chunk| (chunk[0].clone(), chunk[1].clone()))
+            .unzip();
+
+        let keys = make_array(keys);
+        let values = make_array(values);
```

Comment:
I want to invoke the `make_array` to do the aggregation. That's why I put the implementation in `functions-array`.

Reply:
Ideally i think this should be implemented in `functions` inside `core`. 
Do we have any downside of adding `functions-array` as depedency to `functions` ?

Reply:
Maybe we could move `make-array` to functions core-feature?

Reply:
> Do we have any downside of adding functions-array as depedency to functions

Then you need to import unnecessary array function crate if you only care about `funcitons`

Reply:
I guess we can reuse `make_array_inner` if we move `make_array` to `functions` crate.

The alternative is to keep the code here in `functions-array`

Reply:
Yes, I think moving `make_array_inner` to functions is a good idea. It would be beneficial for many scenarios.

Reply:
Hmm, okay. After some research, I believe it's not easy to move `make_array` to `functions`. It's tied to methods in `utils.rs` and `macro.rs`. Moving all the required methods to `functions` could make the codebase chaotic. For now, I prefer to keep them in `functions-arrays`.